pkg/cmd/pulumi-language-java/main.go

```diff
@@ -391,3 +393,59 @@ func (host *javaLanguageHost) InstallDependencies(req *pulumirpc.InstallDependen
 	logging.V(5).Infof("InstallDependencies(Directory=%s): done", req.Directory)
 	return nil
 }
+
+func (host *javaLanguageHost) GetProgramDependencies(
+	ctx context.Context, req *pulumirpc.GetProgramDependenciesRequest) (*pulumirpc.GetProgramDependenciesResponse, error) {
+	// TODO: Implement dependency fetcher for Java
```

Comment:
note: might be worth to create a ticket for this (if not present)

I think implementation in `GetRequiredPlugins` could be reused here.

Reply:
https://github.com/pulumi/pulumi-java/issues/770